### ISSUE TITLE
RunwiseFIR looks upside-down?

### ISSUE THREAD

**Issue by jjzhang-uestc** [cvnlab/GLMsingle#141]
We are new to GLMsingle and recently applied it to an old dataset preprocessed using SPM12. The runwiseFIR results looked fine. 
![runwiseFIR](https://github.com/cvnlab/GLMsingle/assets/86288211/605761ed-b2cb-48f2-a608-46dddf6990d6)

However, we then preprocessed the raw data using the standard fMRIPrep pipeline with all default parameters. After applying GLMsingle to the fMRIPrep-processed data, we noticed that the runwiseFIR results appeared upside-down. 
![runwiseFIR](https://github.com/cvnlab/GLMsingle/assets/86288211/ef458a92-1808-4119-82e5-743c4698c267)

Could you help us understand what this upside-down FIR might indicate? We would greatly appreciate your assistance. Thank you!

**Comment by kendrickkay** [comment]
Is it possible for you to share the entire set of figures for both cases?

What's probably happening is a difference in voxel selection.  The FIR figure summarizes FIR results averaged across a subset of voxels.  Probably, due to incidental differences in the two preprocessed datasets, different voxels are getting selected.  The figures will help diagnose what's goin gon.

**Comment by jjzhang-uestc** [comment]
Thank you for your help! Since there are a number of figures. I uploaded two zip-files that contain the figures for both cases.



> Is it possible for you to share the entire set of figures for both cases?
> 
> What's probably happening is a difference in voxel selection. The FIR figure summarizes FIR results averaged across a subset of voxels. Probably, due to incidental differences in the two preprocessed datasets, different voxels are getting selected. The figures will help diagnose what's goin gon.

**Comment by kendrickkay** [comment]
I took a look.  A few observations:

- betaviz_typeB.png in the fmriprep version looks very strange (and likely "bad").  TypeB is just a simple model with an assumed HRF and a few low frequency polynomial drift terms.  Perhaps is there some confound/drift removal happening in the preprocessing?  I would recommend omitting those steps.  More about this issue is at https://glmsingle.readthedocs.io/en/latest/wiki.html

- If you compare runwiseFIR_summaryvoxels.png across the two versions, you see that the results are substantially dfiferent.  This is probably what is giving rise to the FIR results looking so different.  Also, note that the SPM results appear to be better.  This may be related to the fact that the typeB results (as mentioned above) are strange.

- Another clue comes from the onoffR2.png results, which look better in the SPM version.  The difference between the SPM and fmriprep results with respect to onoffR2 are much larger than expected.  Again, this might all be due to some aggressive confound removal in fmriprep.

So, one hypothesis is that some sort of confound removal is happening and this is affecting many different diagnostic analyses: (typeB, FIR, onoffR2).

**Comment by jjzhang-uestc** [comment]
Thank you for your valuable suggestions!

I think the problem was solved. It was because of the dummy scans (first 5 TRs). FMRIPrep does not automatically throw away the dummy scans. Instead, they generate confounds for them. But fMRIPrep only generates the confounds but does not do the confound removal in the pre-processing. 

So what I did is simply remove the dummy scans and adjust the design matrix. And now the results look much normal.
runwiseFIR:
![runwiseFIR](https://github.com/cvnlab/GLMsingle/assets/86288211/d276aecf-44e2-4f6d-83fe-14119daf5a8c)
dmetric_typeB
![betaviz_typeB](https://github.com/cvnlab/GLMsingle/assets/86288211/37d117fd-e434-4fae-a59b-d9d5917a003a)

Again thank you for your help, we really appreciate it!

**Comment by jjzhang-uestc** [comment]
Thank you for your valuable suggestions!

I think the problem was solved. It was because of the dummy scans (first 5 TRs). FMRIPrep does not automatically throw away the dummy scans. Instead, they generate confounds for them. But fMRIPrep only generates the confounds but does not do the confound removal in the pre-processing. 

So what I did is simply remove the dummy scans and adjust the design matrix. And now the results look much normal.
runwiseFIR:
![runwiseFIR](https://github.com/cvnlab/GLMsingle/assets/86288211/d276aecf-44e2-4f6d-83fe-14119daf5a8c)
dmetric_typeB
![betaviz_typeB](https://github.com/cvnlab/GLMsingle/assets/86288211/37d117fd-e434-4fae-a59b-d9d5917a003a)

Again thank you for your help, we really appreciate it!
